docs/docs/topics/computed-attributes.mdx

```diff
@@ -0,0 +1,42 @@
+---
+title: Computed attributes
+---
+
+The computed attributes feature enables the dynamic calculation of attribute values based on user-defined logic. Currently, Infrahub supports two main types of computed attributes: Jinja2 and Python.
+
+## Jinja2 Computed Attributes
+
+Users can input a concise Jinja2 template directly within the schema definition. The logic provided in the Jinja2 code determines the computed value.
+
+- The computation takes place synchronously within the same transaction. This means any change to the field's input will trigger an immediate update of the computed field.
+- Suitable for use in mandatory attributes because of its immediate processing nature.
+
+See the [guide](/guides/computed-attributes) for instructions on creating Jinja2 computed attribute in Infrahub.
+
+### Limitations
+
+- Jinja2 computed attributes cannot reference relationships with cardinality “many”.
+- Only direct relationships can be used (e.g., a relationship of a relationship is not accessible).
+
+## Python Computed Attributes
+
+Python computed attributes leverage Infrahub's [Python transform](./transformation.mdx) feature. Users can define which transformation to apply directly within the schema.
+
+- Computations are executed asynchronously, meaning there might be a delay before the computed value updates.
+- Python scripts do not have the limitations of Jinja2 computed attributes and can include more complex logic, including relationships with cardinality “many” and nested relationships.
+
+See the [guide](/guides/computed-attributes) for instructions on creating Python computed attribute in Infrahub.
+
+### Limitations
+
+- Python computed attributes cannot be used on mandatory attribute.
+- Complex scripts can impact system performance.
+
+## Choosing Between Jinja2 and Python
+
+Computed attribute kind | Processing Mode   | Mandatory attributes | Relationship limitations                  | Complexity Handling
+------------------------|-------------------|----------------------|-------------------------------------------|---------------------
+Jinja2                  | Synchronous       | Compatible           | No use of “many” cardinality; direct only | Simple operations
```

Comment:
The processing mode here isn't quite correct. For Jinja2 we will try to use sync mode on create, this is why we can have a mandatory attribute here as we manage this logic internally. It could potentially also be possible to update an attribute on the directly connected node in sync mode but overall this always happens async. Without this a change to a related node that in turn requires regeneration of multiple attributes on related nodes could potentially take a long time to complete.